src/sage/rings/number_field/order.py

```diff
@@ -2082,7 +2082,16 @@ def _repr_(self):
                     s = "Gaussian Integers"
         else:
             s = "Order"
-        return s + " in " + repr(self._K)
+            if self.number_field().absolute_degree() == 2:
+                f = (self.discriminant() / self.number_field().discriminant()).sqrt()
+                s += f' of conductor {f}'
+        try:
+            gens = self.ring_generators()
+        except TypeError:
+            gens = list(self.gens())
+        s += f' generated by {gens[0] if len(gens) == 1 else gens}'
```

Comment:
Do we need to convert to a list here? Perhaps it's for consistency with `self.ring_generators()` which returns a tuple. 

Instead of converting and creating a new list, you may just print it by adding '[' and ']' on both sides in L2092 if `len(gens) > 1`.

Reply:
This is a minor point. You may ignore this comment if you will.

Reply:
By the way, you may only print the generators without "[...]".

Reply:
Thus `' generated by {}'.format(', '.join(gens))` suffices.

Reply:
Personally I would keep the brackets since those lists can get quite long for general number fields and the brackets are making it easier to parse visually. If you have a strong opinion about this let me know; I don't.

Reply:
OK, if you like it.